salt/beacons/status.py

```diff
@@ -116,9 +117,11 @@ def beacon(config):
             'time': 'all',
         }
 
-    ret = {}
     for func in config:
-        data = __salt__['status.{0}'.format(func)]()
+        try:
+            data = __salt__['status.{0}'.format(func)]()
+        except (KeyError, AttributeError):
+            continue
```

Comment:
I don't think this is the behavior that we want. This will just cause the beacon to no-op without any indication of what's wrong. A better approach here might be to detect if the pre-requisites are available in the beacon's `__virutal__` and simply not load the beacon if they are not met?